Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -52,48 +53,273 @@ public static class CustomEntitySearch
             }
 
             WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
-                (inRecord, outRecord) => {
-                    string text = inRecord.Data["text"] as string;
-                    IList<string> words = ((JArray)inRecord.Data["words"]).ToObject<List<string>>();
-                    if (words == null)
+                 (inRecord, outRecord) => {
+                     if (!inRecord.Data.ContainsKey("text"))
+                     {
+                         outRecord.Errors.Add(new WebApiErrorWarningContract { Message = "The given key 'text' was not present in the dictionary." });
+                     }
+                     else if (!inRecord.Data.ContainsKey("words") && 
+                     (inRecord.Data.ContainsKey("synonyms") || inRecord.Data.ContainsKey("exactMatches") || inRecord.Data.ContainsKey("fuzzyMatchOffset")))
+                     {
+                         outRecord.Errors.Add(new WebApiErrorWarningContract { Message = "Cannot process record without the given key 'words' in the dictionary"});
+                     }
+                     else
+                     {
+                         string text = inRecord.Data["text"] as string;
+                         IList<string> words = (inRecord.Data.ContainsKey("words")) ? 
+                         ((JArray)inRecord.Data["words"]).ToObject<List<string>>() : new List<string>();
+                         Dictionary<string, string[]> synonyms = (inRecord.Data.ContainsKey("synonyms")) ? 
+                         ((JContainer)inRecord.Data["synonyms"]).ToObject<Dictionary<string, string[]>>() : new Dictionary<string, string[]>();
+                         IList<string> exactMatches = (inRecord.Data.ContainsKey("exactMatches")) ? 
+                         ((JArray)inRecord.Data["exactMatches"]).ToObject<List<string>>() : new List<string>();
+                         int offset = (inRecord.Data.ContainsKey("fuzzyMatchOffset") && (int)(long)inRecord.Data["fuzzyMatchOffset"] >= 0) ? 
+                         (int)(long)inRecord.Data["fuzzyMatchOffset"] : 0;
+                         bool caseSensitive = (inRecord.Data.ContainsKey("caseSensitive")) ? (bool)inRecord.Data.ContainsKey("caseSensitive") : false;
+                         if (words.Count == 0 || (words.Count == 1 && words[0] == ""))
+                         {
+                             outRecord.Warnings.Add(new WebApiErrorWarningContract { Message = "The given key 'words' was not present in the dictionary." });
```

Comment:
>The given key 'words' was not present in the dictionary. [](start = 96, length = 56)

nit: this warning doesn't help the user understand what the implications of not having words parameter present is. Maybe like "Used predefined key words from customLookupSkill configuration file since no 'words' parameter was supplied in web request"

Reply:
Is it better to have a warning or none at all?

Reply:
I think none at all but you could get a second opinion on this.